wagtailio/navigation/templatetags/navigation_tags.py

```diff
@@ -0,0 +1,54 @@
+from django import template
+
+register = template.Library()
+
+
+@register.inclusion_tag("patterns/includes/menu_footer.html", takes_context=True)
+def footer_navigation(context):
+    try:
+        footer_navigation = context["settings"]["navigation"][
+            "NavigationSettings"
+        ].footer_navigation
+    except (KeyError, AttributeError):
+        footer_navigation = None
+
+    return {
+        "footer_navigation": footer_navigation,
+    }
+
+
+@register.inclusion_tag("patterns/includes/menu_main.html", takes_context=True)
+def main_navigation(context):
+    menus = []
+
+    try:
+        main_navigation = context["settings"]["navigation"][
+            "NavigationSettings"
+        ].main_navigation
+
+        for item in main_navigation.menu_sections.all():
```

Comment:
Suggestion: You could replace this for-loop with extend instead; 

```
menus.extend({"name": item.menu_section.name, "nav_items": item.menu_section.nav_items, "call_to_action": item.menu_section.call_to_action,} for item in main_navigation.menu_sections.all()
```

Reply:
reformatted:

```python
menus.extend(
    {
        "name": item.menu_section.name,
        "nav_items": item.menu_section.nav_items,
        "call_to_action": item.menu_section.call_to_action,
    }
    for item in main_navigation.menu_sections.all()
)
```

which is very close the for-loop and I don't see any gains in readability or other? 🤔

Reply:
I did my own test:

```python
import timeit
from faker import Faker

fake = Faker()

menu_sections = [
    {
        "name": fake.name(),
        "call_to_action": fake.sentence(),
        "nav_items": fake.words(),
    }
    for _ in range(40)
]


def append_fn():
    menus = []
    for item in menu_sections:
        menus.append(item)


def extend_fn():
    menus = []
    menus.extend(item for item in menu_sections)


print(min(timeit.repeat(lambda: append_fn())))
print(min(timeit.repeat(lambda: extend_fn())))
```

and `append` wins, although the difference is small.

```console
❯ python append_vs_extend.py
2.6496993280015886
2.758143986982759
❯ python append_vs_extend.py
2.60596901900135
2.7336814689915627
```